themes/bootstrap3/js/check_item_statuses.js

```diff
@@ -116,6 +126,8 @@ function itemQueueAjax(id, el) {
   itemStatusEls[id] = el;
   itemStatusTimer = setTimeout(runItemAjaxForQueue, itemStatusDelay);
   el.addClass('js-item-pending').removeClass('hidden');
+  el.find('.callnumAndLocation').removeClass('hidden');
+  el.find('.callnumAndLocation .ajax-availability').removeClass('hidden');
   el.find('.status').removeClass('hidden');
```

Comment:
This shouldn't hold back this PR, but it seems that this logic needs some revisiting. Are we really hiding an element (`.ajax-availability`) inside a hidden element (`.callnumAndLocation`) inside a hidden element (`.status`) inside a hidden element (`.record`)?

Reply:
Yes, it's pretty bad and should be revisited, but I think it's out of scope for this PR..

Reply:
Though, to be frank, there are reasons for the complexity since the mechanism supports so many different ways of reporting the availability. I think it would make sense to revisit the whole thing and perhaps move to a mechanism that quite simply returns an html snippet for each record.

Reply:
I would definitely be in favor of revisiting this, as I think it can probably be simplified. However, I'm not sure we can go so far as "just return an HTML snippet" without doing duplicate work, since the HTML is a combination of data from search backends and from the ILS. I wouldn't want to have to re-query the backend in addition to querying the ILS in order to resolve the AJAX call... but because of the way the HTML is currently structured, it's hard to isolate the ILS-specific stuff without some degree of complexity.

That being said, whenever anything has organically grown for a decade (which is really the case with this code), it can probably be simplified with a ground-up rethink.

Reply:
Maybe I'm missing something (again), but if the driver supports ajax availability, the content of div.callnumAndLocation (https://github.com/vufind-org/vufind/blob/master/themes/bootstrap3/templates/RecordDriver/DefaultRecord/result-list.phtml#L102) all comes from getItemStatuses, no? In that case getItemStatuses could simply return whatever is needed based on settings in Item_Status section. The call numbers from driver are used only if ajax availability is not supported, right?